shadow-dom/slotchange-customelements.html

```diff
@@ -0,0 +1,44 @@
+<!DOCTYPE html>
+<html>
+<head>
+<title>Shadow DOM: slotchange customelements</title>
+<meta name="author" title="Surma" href="mailto:surma@google.com">
+<link rel="help" href="https://dom.spec.whatwg.org/#signaling-slot-change">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+</head>
+<body>
+<my-custom-element id="upgrade"><div></div></my-custom-element>
+<script>
+var calls = [];
+class MyCustomElement extends HTMLElement {
+    constructor() {
+        super();
+    }
+
+    connectedCallback() {
+        this.attachShadow({mode: 'open'});
+        this.shadowRoot.innerHTML = '<slot></slot>';
+        var slot = this.shadowRoot.children[0];
+        slot.addEventListener('slotchange', function() {
+            calls.push(this);
```

Comment:
This might be better as `calls.push(this.id)`. Then below you can do a single test, something like `assert_array_equals(calls, ["upgrade", "parser"])`. That has the bonus of testing that things are in the correct order.